woocommerce/woocommerce-gateway-payfast
# Decode HTML entity encoding in the site name before sending data to Payfast.

### All Submissions:



---

### Changes proposed in this Pull Request:
The PR addresses the signature mismatch error. Upon investigation, I discovered that the mismatch was due to HTML entity encoding in the site/blog name for special characters. We are sending the blog name in the item name and item description values, and the signature is being generated based on the encoded HTML entity. However, the value in the form is sent with decoded HTML entity, causing the mismatch. This PR resolves the issue by generating the signature based on the decoded HTML entity.

### Steps to test the changes in this Pull Request:
1. Set the site title to something that uses a special character, e.g., `Bob's Site` (Please try different combinations).
2. Add a product to the cart and place an order.
3. Verify that the order has been placed successfully without any issues.

### Changelog entry
> Fix - Resolved signature mismatch error caused by HTML entity encoding in site/blog name.
